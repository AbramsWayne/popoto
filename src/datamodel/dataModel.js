var dataModel = {};

dataModel.nodes = [];
dataModel.links = [];

export default dataModel