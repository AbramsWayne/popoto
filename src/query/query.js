var query = {};
export default query