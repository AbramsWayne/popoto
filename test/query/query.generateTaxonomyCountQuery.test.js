import provider from "../../src/provider/provider";
import query from '../../src/query/query.js'
import dataModel from "../../src/datamodel/dataModel";

describe("taxonomy count generation", function () {
    beforeEach(() => {
        provider.node.Provider = {
            "Person": {
                returnAttributes: ["name", "born"],
                constraintAttribute: "name",
            },
            "Movie": {
                returnAttributes: ["title", "born"],
            }
        };
    });

    afterEach(() => {
        delete provider.node.Provider;
    });

    test("Person", () => {
        var countQuery = query.generateTaxonomyCountQuery("Person");
        expect(countQuery).toMatchSnapshot();
    });
    test("Movie", () => {
        var countQuery = query.generateTaxonomyCountQuery("Movie");
        expect(countQuery).toMatchSnapshot();
    });
});