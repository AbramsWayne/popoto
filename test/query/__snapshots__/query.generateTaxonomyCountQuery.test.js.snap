// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`taxonomy count generation Movie 1`] = `"MATCH (n:\`Movie\`) RETURN count(DISTINCT ID(n)) as count"`;

exports[`taxonomy count generation Person 1`] = `"MATCH (n:\`Person\`) RETURN count(DISTINCT n.name) as count"`;
